porcupine/plugins/text_google_search.py

```diff
@@ -0,0 +1,17 @@
+"""Search selected text on Google."""
+
+import tkinter as tk
+from porcupine import menubar, tabs
+import webbrowser
+import urllib.parse
+
+def google_search(tab: tabs.FileTab) -> None:
+    try:
+        selected_text = tab.textwidget.get("sel.first", "sel.last")
+        url = "https://www.google.com/search?q={}".format(urllib.parse.quote_plus(selected_text))
+        webbrowser.open_new_tab(url)
+    except:
+        pass
```

Comment:
- It would be good to specify the exception being caught here, as in `except TclError:`.
- Avoid putting many lines of code inside try/except. Only put the first line, because that's where the error will come from.

Something like this:

```
try:
    selected_text = ...
except tk.TclError:
    return

url = ...
webbrowser.open_new_tab(url)
```

Reply:
It might be good to also add a comment like `# nothing selected` or similar. It won't be obvious for someone else looking at the code :)

Reply:
resolved in commit https://github.com/Akuli/porcupine/pull/1024/commits/b7105b4ff877b0fef23a4df394a50a879c1c701e